app/controllers/books_controller.rb

```diff
@@ -1,5 +1,6 @@
 class BooksController < ApplicationController
   before_action :require_login, only: %i[new create edit update destroy]
+  before_action :require_moderator, only: %i[new create edit update destroy]
```

Comment:
isn't the require_login made redundant here since require_moderator also checks if the user is logged_in?

Reply:
Require login ensures people are authenticated, redirecting them to login.
Require moderator just breaks a 401

We need both because we want the redirect part + the 401 part afterward.